cea/optimization_new/energyPotential.py

```diff
@@ -0,0 +1,193 @@
+        average_return_temperature = self._get_building_potentials(scfp_potential_files, building_codes,
+                                                                  'Q_SC_gen_kWh', 'T_SC_re_C')
+        self.main_energy_carrier = self._temp_to_thermal_ec(average_return_temperature, thermal_energy_carriers)
+        return self
+
+    def load_geothermal_potential(self, geothermal_potential_file, thermal_energy_carriers):
+        self.type = 'Geothermal'
+        self.scale = 'Domain'
+        if exists(geothermal_potential_file):
+            geothermal_potential = pd.read_csv(geothermal_potential_file)
+            self.main_potential = geothermal_potential.QGHP_kW
+            average_return_temperature = self._get_average_temp(geothermal_potential.Ts_C)
+            self.main_energy_carrier = self._temp_to_thermal_ec(average_return_temperature, thermal_energy_carriers)
+            return self
+        else:
+            return None
+
+    def load_water_body_potential(self, water_body_potential_file, thermal_energy_carriers):
+        self.type = 'WaterBody'
+        self.scale = 'Domain'
+        if exists(water_body_potential_file):
+            water_body_potential = pd.read_csv(water_body_potential_file)
+            self.main_potential = water_body_potential.QLake_kW
+            average_return_temperature = self._get_average_temp(water_body_potential.Ts_C)
+            self.main_energy_carrier = self._temp_to_thermal_ec(average_return_temperature, thermal_energy_carriers)
+            return self
+        else:
+            return None
+
+    def load_sewage_potential(self, sewage_potential_file, thermal_energy_carriers):
+        self.type = 'Sewage'
+        self.scale = 'Domain'
+        if exists(sewage_potential_file):
+            sewage_potential = pd.read_csv(sewage_potential_file)
+            self.main_potential = sewage_potential.Qsw_kW
+            average_return_temperature = self._get_average_temp(sewage_potential.Ts_C)
+            self.main_energy_carrier = self._temp_to_thermal_ec(average_return_temperature, thermal_energy_carriers)
+            return self
+        else:
+            return None
+
+    def _get_building_potentials(self, energy_potential_files, building_codes, main_potential_column_name,
+                                 temperature_column_name=None, auxiliary_potential_column_name=None):
+        """
+        Gets main and auxiliary potentials from the stored energy potential files and stores them in the corresponding
+        object attributes. In case a temperature column name is indicated, the average temperature (when operating
+        the corresponding component at maximum potential) is returned.
+        """
+        # check if there are potential files for any of the buildings
+        if not any([exists(file) for file in energy_potential_files]):
+            print(f"No {self.type} potentials could be found for the indicated buildings.")
```

Comment:
Maybe add. "Consider running potentials scripts" or something like that (CEA has some warnings like that which are useful when another script can solve your problem.)

Reply:
Good point. Done.